crates/core/c8y_api/src/utils.rs

```diff
@@ -13,6 +14,38 @@ pub mod bridge {
             Err(_err) => false,
         }
     }
+
+    pub fn is_c8y_bridge_established(message: &Message) -> bool {
+        match message.payload_str() {
+            Ok(payload) => {
+                message.topic.name == C8Y_BRIDGE_HEALTH_TOPIC
+                    && (payload == C8Y_BRIDGE_UP_PAYLOAD || payload == C8Y_BRIDGE_DOWN_PAYLOAD)
+            }
+            Err(_err) => false,
+        }
+    }
+}
+
+pub mod tedge_agent {
+    use mqtt_channel::Message;
+
+    pub const TEDGE_AGENT_HEALTH_TOPIC: &str = "tedge/health/tedge-agent";
+    const TEDGE_HEALTH_UP_PAYLOAD: &str = r#""status":"up""#;
+    const TEDGE_HEALTH_DOWN_PAYLOAD: &str = r#""status":"down""#;
+
+    pub fn check_tedge_agent_status(message: &Message) -> bool {
+        if message.topic.name.eq(TEDGE_AGENT_HEALTH_TOPIC) {
+            match message.payload_str() {
+                Ok(payload) => {
+                    payload.contains(TEDGE_HEALTH_UP_PAYLOAD)
+                        || payload.contains(TEDGE_HEALTH_DOWN_PAYLOAD)
+                }
```

Comment:
Again, I don't really see the point of checking the payload content as we're fine with both `up` and `down` statuses. So, just checking the topic is more than sufficient.

Reply:
I thought its better to check the `payload` as well, as there are chances of having the wrong payload.